core/trino-spi/src/main/java/io/trino/spi/connector/FixedPageSource.java

```diff
@@ -27,14 +28,21 @@
     private boolean closed;
 
     public FixedPageSource(Iterable<Page> pages)
+    {
+        this(pages, Optional.empty());
+    }
+
+    public FixedPageSource(Iterable<Page> pages, Optional<Long> fixedMemoryUsageBytes)
     {
         this.pages = pages.iterator();
 
-        long memoryUsageBytes = 0;
-        for (Page page : pages) {
-            memoryUsageBytes += page.getRetainedSizeInBytes();
-        }
-        this.memoryUsageBytes = memoryUsageBytes;
+        this.memoryUsageBytes = fixedMemoryUsageBytes.orElseGet(() -> {
```

Comment:
Between this method not being right for us, and it not having a `close` method, maybe we shouldn't be trying to reuse `FixedPageSource`

Reply:
Created `io.trino.plugin.iceberg.IncrementalPopulatingPageSource`